bots/s3_only_allow_ssl.py

```diff
@@ -0,0 +1,178 @@
+
+    return res
+
+
+# The function find which actions should add to policy and return the missing Effects in a json format
+def get_actions(bucket_name, account_number, options):
+    if ("*" in options) or ("s3:*" in options) or ('s3:GetObject' in options and 's3:PutObject' in options) or (
+            's3:Get*' in options and 's3:Put*' in options):
+        Effect = {
+            "Effect": "Deny",
+            "Principal": "*",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*",
+            "Condition": {
+                "Bool": {
+                    "aws:SecureTransport": "false"
+                }
+            }
+        }
+        return Effect, None
+
+    elif 's3:GetObject' in str(options) or "s3:Get*" in str(options):
+        first_Effect = {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": account_number
+            },
+            "Action":
+                's3:PutObject',
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*"
+        }
+        second_Effect = {
+            "Effect": "Deny",
+            "Principal": "*",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*",
+            "Condition": {
+                "Bool": {
+                    "aws:SecureTransport": "false"
+                }
+            }
+        }
+        return first_Effect, second_Effect
+    elif 's3:PutObject' in str(options) or "s3:Put*" in str(options):
+        first_Effect = {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": account_number
+            },
+            "Action":
+                's3:GetObject',
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*"
+        }
+        second_Effect = {
+            "Effect": "Deny",
+            "Principal": "*",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*",
+            "Condition": {
+                "Bool": {
+                    "aws:SecureTransport": "false"
+                }
+            }
+        }
+        return first_Effect, second_Effect
+    else: # need to add both s3:PutObject and s3:GutObject
+        first_Effect = {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": account_number
+            },
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*",
+        }
+        second_Effect = {
+            "Effect": "Deny",
+            "Principal": "*",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::" + bucket_name + "/*",
+            "Condition": {
+                "Bool": {
+                    "aws:SecureTransport": "false"
+                }
+            }
+        }
+        return first_Effect, second_Effect
+
+
+# The Function return the SSL policy if the bucket does not have a bucket policy
+def make_new_policy(bucket_name, account_number):
```

Comment:
You can move this object into constant and get rid of the function

Reply:
make it constant